src/super_gradients/training/losses/ohem_ce_loss.py

```diff
@@ -52,3 +53,21 @@ def forward(self, logits, labels):
         else:
             loss = loss[:num_mining]
         return torch.mean(loss)
+
+
+class OhemBCELoss(OhemCELoss):
+    """
+    OhemBCELoss - Online Hard Example Mining Binary Cross Entropy Loss
+    """
+    def __init__(self,
```

Comment:
Binary cross entropy can be used with multi-classes not only with binary tasks. In case of multi-classes the target tensor is with shape [BxCxHxW], and usually a one hot tensor.

Reply:
Yes thats one of the changes I wanted to make.
Ill change the class hierarchy so that we have OhemLoss which gets the criteria as an argument in the constructor, then also allow to pass a different criteria to StdcLoss as well

Reply:
In case of multi-classes we should keep the option of ignore labels

Reply:
fixed